### PR TITLE
add payment plan

### PR DESCRIPTION
**Pull request by mgamarrav**
1. redux slice was added in redux/slice
2. was added a new state in redux/store
3. was created TableCompute.js in creditDetails.js in app/pages/compute
4. new validations was added to SchemaValidation.js
5. was installed a new library momentJS to handle dates
6. Many files was affected because I moved the button componnet to common file

### CODE REVIEW

File: src/app/utils/getDate.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const dateList = [];
+  const startingDay = moment(startDay);
+  
+  [...Array(Number(monthNumber))].map((_, index) => {
+    const nextDate = startingDay.clone().add(index, 'months');
+    let payDay = startingDay.date();
+    const nextMonth = nextDate.month();
+    const nextYear = nextDate.year();
+
+    var lastDayOfMonth = moment([nextYear, nextMonth]).endOf('month').date();
```
Comment (douglascamposh) [inline src/app/utils/getDate.js]: please use cons or let instead of

File: src/app/utils/validations/schemaValidation.js [hunk 2]
```diff
@@ -138,3 +138,24 @@ export const validationRolScheme = () => {
...
+  downPayment: Yup.number()
+    .required("El monto de aporte inicial es requerido")
+    .moreThan(0, "El monto debe ser mayor que cero")
+    .test('decimal-places', 'El precio debe tener máximo dos decimales', value => {
+      if (!value) return true;
+      return /^\d+(\.\d{1,2})?$/.test(value.toString());
+    }),
+  months: Yup.number()
+    .required("El plazo es requerido"),
+  setDate: Yup.date()
```
Comment (douglascamposh) [inline src/app/utils/validations/schemaValidation.js:159]: this line is working? seems in the schema this field is called startDate, please take a look

File: src/app/(pages)/compute/page.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            </Card>
+          )
+        }
+      </div>
+    </div>
+  );
+}
+
+const styles = {
+	containerStyle: `
```
Comment (douglascamposh) [inline src/app/(pages)/compute/page.js]: please fix the indentation

File: src/app/utils/validations/schemaValidation.js (hunk 2 above)
Comment (mgamarrav) [reply src/app/utils/validations/schemaValidation.js:159]: yep it is working correctly in the Field component the name was established as setDate

File: src/app/(pages)/compute/page.js (hunk 3 above)
Comment (mgamarrav) [reply src/app/(pages)/compute/page.js]: this identation is very extrange because in my code editor I just have 2 spaces but just in case I formated code

**Review by douglascamposh — APPROVED** [review]

**Pull request merged**